source/extensions/resource_monitors/envoy_container_cpu_utilization/container_stats_reader.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <chrono>
+#include "source/common/common/logger.h"
+#include "source/common/common/utility.h"
+
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace EnvoyContainerCpuUtilizationMonitor {
+
+
+struct EnvoyContainerStats {
```

Comment:
Can we fold this into the existing cpu utilization monitor?

For example if we were to modify the proto to have an enum or WKT for the reader to instantiate? 

we could then modify:
https://github.com/envoyproxy/envoy/blob/e22980c406bc6cad8e3545343fe8a503f47bf329/source/extensions/resource_monitors/cpu_utilization/config.cc#L19 

to create a cpu monitor that uses either one of the stat readers ( the container one or the host one)

This would lead to us having less code to own in general :).